intel-sgx/enclave-runner/src/command.rs

```diff
@@ -63,6 +63,13 @@ impl Command {
     }
 
     pub fn run(self) -> Result<(), Error> {
-        EnclaveState::main_entry(self.main, self.threads, self.usercall_ext, self.forward_panics, self.cmd_args)
+        EnclaveState::main_entry(self.main, self.threads, self.usercall_ext, self.forward_panics, self.cmd_args, num_cpus::get())
+    }
+
+    /// Run the enclave with the provided number of worker threads.
+    /// # Panics
```

Comment:
Why is it a panic if the number of worker threads is 0 and not an error?

Reply:
I can make it an error. But my thinking is that `worker_threads == 0` is a user error/precondition failure that the user must fix and not an error that comes out of running the enclave.

Another option would be to use `NonZeroUsize`, but it's very rarely used in Rust libraries.

Reply:
Libraries should try really hard to avoid panics. Panics in libraries should really only be for fatal conditions that the library has no way to recover from, like internal data structure inconsistencies that can't be reconciled somehow. An incorrect number of worker threads is absolutely not fatal in this way. Either the API should make it impossible to pass 0 via `NonZeroUsize`, or the call should return an error. The caller could decide it wants to panic if it gets this error, or it might try to make the call again with a different number of threads.

I think having an error for an invalid number of threads makes sense, as we probably want to have some upper bound on the number of threads as well. Does it really make sense for the application to request for example trillions of threads?

Reply:
I'm now returning an error instead of panicking when `num_worker_threads` is set to 0.